core/generated.go

```diff
@@ -616,7 +620,7 @@ func encapsulateMutator(mctx blueprint.TopDownMutatorContext) {
 	}
 
 	mctx.WalkDeps(func(child blueprint.Module, parent blueprint.Module) bool {
-		if mctx.OtherModuleDependencyTag(child) != generatedDepTag {
+		if mctx.OtherModuleDependencyTag(child) != encapsulationTag {
 			return false
 		}
 		parentProp, pok := getGenerateCommon(parent)
```

Comment:
We should only have encapsulationTags on generated sources, because we only added them to the correct module types.

This means we can remove the checks on line 630 and 633.

Reply:
(Not sure if github's showing this correctly, but this comment only applies to the second commit!)